### DISCUSSION TITLE
Moving builds property to functions in vercel.json

### DISCUSSION THREAD

**Discussion by weiquu** [vercel/community#4518] [Help]
### Summary

I am trying to deploy a Python API with the `maxDuration` increased. However, removing the `build` property in `vercel.json` causes the deployment to fail.

### Steps to Reproduce

Here is my original `vercel.json` file:

```
{
    "builds": [
        {
            "src": "/api/main.py",
            "use": "@vercel/python"
        }
    ],
    "routes": [
        {
            "src": "/(.*)",
            "dest": "api/main.py"
        }
    ]
}
```

It deploys fine, but I want to increase the `maxDuration`. To do so, I added the `functions` property:
```
    "functions": {
        "api/**/*": {
            "maxDuration": 200
        }
    }
```

Because the `functions` property cannot be used in conjunction with the `builds` property, I removed the `builds` property. However, now the deployment fails (times out because the `uvicorn api.main:app --host 0.0.0.0 --port 3000` command runs for very long). How should I "convert" the `builds` property to the `functions` property?

**Comment by swarnava** [comment]
Hi,

What error are you receiving when you use both `builds` and `functions` property ? It should work like this:

```js
{
    "version": 2,
    "name": "foodManagment",
    "builds": [{
        "src": "./index.js",
        "use": "@vercel/node"
    }],
    "routes": [{
        "src": "/(.*)",
        "dest": "/"
    }],
    "functions": {
        "api/categories.js": {
            "memory": 1024,
            "maxDuration": 60
        },
        "api/client.js": {
            "memory": 1024,
            "maxDuration": 60
        },
        "api/dish.js": {
            "memory": 1024,
            "maxDuration": 60
        },
        "api/table.js": {
            "memory": 1024,
            "maxDuration": 60
        }
    }
}
```

**Reply by weiquu** [reply]
Hi @swarnava, this is the error I'm seeing:

Error: The `functions` property cannot be used in conjunction with the `builds` property. Please remove one of them.
Learn More: https://vercel.link/functions-and-builds

**Reply by swarnava** [reply]
Do you have a minimal repository which we can look at?

**Reply by weiquu** [reply]
Sure, [here](https://github.com/weiquu/taleweaver-prod/tree/main/genapi) is what I'm trying to deploy (have set the root directory to `genapi`). As mentioned, it deploys properly without the `functions` property, and I don't think there's a way to set the `maxDuration` anywhere else.

For easy reference, my `vercel.json` settings is in the root directory, and I have an `api` folder that contains the rest of my code.
